django/db/models/sql/compiler.py

```diff
@@ -330,8 +330,21 @@ def _order_by_pairs(self):
                 if not self.query.standard_ordering:
                     field = field.copy()
                     field.reverse_ordering()
+                src = field.get_source_expressions()[0]
+                if isinstance(src, F):
```

Comment:
`field` is always an `OrderBy` instance so we can use `.expression`:
```suggestion
                if isinstance(field.expression, F):
```